PokeDemo/Presenter/HomeViewPresenter.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  HomeViewPresenter.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 15/07/22.
+//
+
+import Foundation
+import UIKit
+
+protocol HomeViewPresenterDelegate: AnyObject {
+    func showPokemonList(pokemons: [Pokemon])
+    func showAlert(title: String, message: String)
+}
+
+typealias HomePresenterDelegate = HomeViewPresenterDelegate & UIViewController
+
+final class HomeViewPresenter {
+    
+    // MARK: Properties
+    
+    // MARK: Public
+    
+    weak var delegate: HomePresenterDelegate?
+    private let errorTitle = "Error"
+
+    // MARK: Private
+    
+    private let service: ServiceRepresentable
+    private var nextURL: String?
+
+    // MARK: Initailizers
+    
+    init(service: ServiceRepresentable = Service.shared) {
+        self.service = service
+    }
+    
+    // MARK: Exposed method(s)
+    
+    func setViewDelegate(delegate: HomePresenterDelegate) {
+        self.delegate = delegate
+    }
+    
+    func isLoadingCell(for indexPath: IndexPath) -> Bool {
+        return nextURL != nil
+    }
+    
+    func getPokemonList() {
+        if let nextURL {
+            self.nextURL = nextURL
+        } else {
+            nextURL = Endpoints.shared.baseURL
+        }
```

Comment:
Remove this, lets do pagination in next PR

Reply:
Done

Reply:
> Done

I don't see any commit number of this change

Reply:
Done in cd97b93468f78a9f35b47f3476ee786dce2aca8a

Reply:
[cd97b93](https://github.com/rudrbansal/Pokemon/pull/2/commits/cd97b93468f78a9f35b47f3476ee786dce2aca8a)